L1Trigger/L1THGCal/plugins/be_algorithms/SingleCellClusterAlgo.cc

```diff
@@ -51,9 +58,22 @@ void SingleCellClusterAlgo::run(const l1t::HGCFETriggerDigiCollection& coll, con
         {
             if(triggercell.hwPt()>0)
             {
+
                 HGCalDetId detid(triggercell.detId());
+                int subdet =  (ForwardSubdetector)detid.subdetId() - 3;
+                if( subdet == 0 ){ 
+                    geom = hgceeGeoHandle.product();     
+                }else if( subdet == 1 ){
+                    geom = hgchefGeoHandle.product();
+                }else if( subdet == 2 ){
+                    //std::cout << "ATTENTION: the BH trgCells are not yet implemented !! "<< std::endl;
+                }
+                const HGCalTopology& topo = geom->topology();
+                const HGCalDDDConstants& dddConst = topo.dddConstants();
+                int cellThickness = dddConst.waferTypeL((unsigned int)detid.wafer() );        
```

Comment:
Related to the previous comment:
The best would probably to retrieve directly the cellThickness without the intermediate (dangerous) pointer.